src/flows/context/api.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  postApiV2privateFlowsOrgsFlow,
+  PostApiV2privateFlowsOrgsFlowParams,
+  PatchApiV2privateFlowsOrgsFlowParams,
+  patchApiV2privateFlowsOrgsFlow,
+  deleteApiV2privateFlowsOrgsFlow,
+  getApiV2privateFlowsOrgsFlows,
+  DeleteApiV2privateFlowsOrgsFlowParams,
+} from 'src/client/flowsRoutes'
+import {notebookUpdateFail} from 'src/shared/copy/notifications'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {v4 as UUID} from 'uuid'
+import {notify} from 'src/shared/actions/notifications'
+
+export const FLOWS_API_FLAG = 'notebooks-api'
+const DEFAULT_API_FLOW: PatchApiV2privateFlowsOrgsFlowParams = {
+  id: '',
+  orgID: '',
+  data: {},
+}
+let stagedFlow: PatchApiV2privateFlowsOrgsFlowParams = DEFAULT_API_FLOW
+let reportDecayTimeout = null
+let reportMaxTimeout = null
+
+const REPORT_DECAY = 2000 // number of miliseconds to wait after last event before sending
+const REPORT_MAX_WAIT = 5000 // max number of miliseconds to wait between sends
+
+export const pooledUpdateAPI = (flow: PatchApiV2privateFlowsOrgsFlowParams) => {
```

Comment:
not for this PR, but i just had to write a simple debounce for my pr too. that means we have one for the query builder, one for notebooks, one for events, and one here. should we make a ticket to make... just.. one?

Reply:
yes and done: https://github.com/influxdata/ui/issues/572